tests/qchem/of_tests/test_convert.py

```diff
@@ -813,3 +813,177 @@ def dummy_ansatz(phis, wires):
+                (4, 4): -0.05003594568491194,
+                (8, 2): 0.04452333085007853,
+                (8, 8): -0.05226230322043741,
+                (16, 16): -0.0404759737476627,
+                (32, 32): -0.0404759737476627,
+            },
+        ),
+    ],
+)
+def test_ucisd_state(molecule, basis, symm, tol, wf_ref):
+    r"""Test that _ucisd_state returns the correct wavefunction."""
+
+    mol = pyscf.gto.M(atom=molecule, basis=basis, symmetry=symm)
+    myhf = pyscf.scf.UHF(mol).run()
+    myci = pyscf.ci.UCISD(myhf).run()
+
+    wf_cisd = qchem.convert._ucisd_state(myci, tol=tol)
+
+    assert wf_cisd.keys() == wf_ref.keys()
+    assert np.allclose(abs(np.array(list(wf_cisd.values()))), abs(np.array(list(wf_ref.values()))))
+
+
+@pytest.mark.parametrize(
+    ("wf_dict", "n_orbitals", "wf_ref"),
+    [
+        (  # -0.99 |1100> + 0.11 |0011>
+            {(1, 1): -0.9942969785398778, (2, 2): 0.10664669927602179},
+            2,
+            np.array(
+                [
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.1066467,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    -0.99429698,
+                    0.0,
+                    0.0,
+                    0.0,
+                ]
+            ),
+        ),
+    ],
+)
+def test_wfdict_to_statevector(wf_dict, n_orbitals, wf_ref):
+    r"""Test that _wfdict_to_statevector returns the correct statevector."""
+    wf_comp = qchem.convert._wfdict_to_statevector(wf_dict, n_orbitals)
+    assert np.allclose(wf_comp, wf_ref)
+
+
+@pytest.mark.parametrize(
+    ("molecule", "basis", "symm", "method", "wf_ref"),
+    [
+        (
+            [["H", (0, 0, 0)], ["H", (0, 0, 0.71)]],
+            "sto6g",
+            "d2h",
+            "ucisd",
+            np.array(
+                [
+                    0.0,
+                    0.0,
+                    0.0,
+                    -0.1066467,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.99429698,
+                    0.0,
+                    0.0,
+                    0.0,
+                ]
+            ),
+        ),
+    ],
+)
+def test_import_state(molecule, basis, symm, method, wf_ref):
+    r"""Test that cisd_state returns the correct wavefunction."""
+
+    mol = pyscf.gto.M(atom=molecule, basis=basis, symmetry=symm)
+    myhf = pyscf.scf.UHF(mol).run()
+    myci = pyscf.ci.UCISD(myhf).run()
+
+    wf_comp = qchem.convert.import_state(myci, method)
+
+    # overall sign could be different in each PySCF run
+    assert np.allclose(wf_comp, wf_ref) or np.allclose(wf_comp, -wf_ref)
```

Comment:
Weird 😆

Reply:
Haha yup, it's annoying! Overall wavefunction signs (phases) are not meaningful here, and in PySCF sometimes different signs come out of calculations, especially with there being variability between methods (CISD vs CCSD, or CCSD vs CASCI, etc.).